### ISSUE TITLE
fail to use docker 1.9.1!

### ISSUE THREAD

**Issue by harryprince** [docker/docker#18893]
docker information
```
sudo docker version
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5
 Built:        Fri Nov 20 17:56:04 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.3
 Git commit:   a34a1d5
 Built:        Fri Nov 20 17:56:04 UTC 2015
 OS/Arch:      linux/amd64
```
```
docker info
Cannot connect to the Docker daemon. Is the docker daemon running on this host?
ubuntu@ip-172-31-21-89:~$ sudo docker info
Containers: 0
Images: 0
Server Version: 1.9.1
Storage Driver: devicemapper
 Pool Name: docker-202:1-150022-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 107.4 GB
 Backing Filesystem:
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 7.342 GB
 Data Space Total: 107.4 GB
 Data Space Available: 2.188 GB
 Metadata Space Used: 7.565 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.14 GB
 Udev Sync Supported: false
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.82-git (2013-10-04)
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 3.13.0-48-generic
Operating System: Ubuntu 14.04.2 LTS
CPUs: 1
Total Memory: 992.5 MiB
Name: ip-172-31-21-89
ID: RBAC:KQDQ:T5PG:66BL:O4AW:DQXH:TNNG:JVUA:4FPM:4L7G:7S3L:TRFT
WARNING: No swap limit support
```

```
uname -a
Linux ip-172-31-21-89 3.13.0-48-generic #80-Ubuntu SMP Thu Mar 12 11:16:15 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux
```
I am using AWS micro-instance.
the comand line information,only one step.
```
sudo docker pull ubuntu
Using default tag: latest
latest: Pulling from library/ubuntu
0bf056161913: Download complete
1796d1c62d0c: Download complete
e24428725dd6: Download complete
89d5d8e8bafb: Download complete
Pulling repository docker.io/library/ubuntu
d55e68e6cc9c: Error pulling image (latest) from docker.io/library/ubuntu, Driver devicemapper failed to create image rootfs 9377ad319b00884df249b7820e3cf540b1c4631b3b1ee6998a0f7c3d53962e03: Error running deviceCreate (createSnapDevice) dm_tError pulling image (latest) from docker.io/library/ubuntu, Driver devicemapper failed to create image rootfs 9377ad319b00884df249b7820e3cf540b1c4631b3b1ee6998a0f7c3d53962e03: Error running deviceCreate (createSnapDevice) dm_task_run failed
```
the daemon information:
```
INFO[2500] POST /v1.21/images/create?fromImage=jenkins%3Alatest
ERRO[2512] filesystem layer verification failed for digest sha256:88468978469dadb134270e3fe4825439eeac4eb652a149660a670ad7648d2e8e
ERRO[2512] filesystem layer verification failed for digest sha256:a1fef06c42d9f689984fd1abab1e775a75a523ac1c4f9bb1c38a3ec2110d40f5
```

**Comment by qq690388648** [comment]
It seems not like a regular problem, does it appear every times?

**Comment by thaJeztah** [comment]
I see a couple of things / have some questions;

- You're running docker with devicemapper on an Ubuntu system. Is there a specific reason you selected devicemapper? In general, aufs is the recommended/default driver on Ubuntu. You may need to install the linux-image-extra packages to add support for aufs, as [explained in the installation docs](https://docs.docker.com/engine/installation/ubuntulinux/#prerequisites-by-ubuntu-version)
- You're running docker **without** Udev Sync support. **This is _known_ to result in data-corruption and data-loss**, so I strongly advice against running in that setup. Read the discussion on https://github.com/docker/docker/pull/14292.
- If you don't want to use aufs, or have specific reasons to use devicemapper, I'd encourage you to not use the default "loop devices", as they are not suitable for production (only to "get things running"). Read the [devicemapper section](https://docs.docker.com/engine/userguide/storagedriver/device-mapper-driver/#configuring-docker-with-device-mapper) in the documentation on how to configure devicemapper for production

How did you install docker? Are you running a statically build binary, or a dynamic binary, installed from our apt repository?

**Comment by thaJeztah** [comment]
Note that you may also be running out of space on the devicemapper device; `Data Space Available: 2.188 GB`

**Comment by unclejack** [comment]
As @thaJeztah has stated above, `Udev Sync Supported: false` is guaranteed to leead to data corruption and data loss.

Since the error you've seen is related to devicemapper and is very likely to be caused by the fact that Docker was probably not installed using the official packages, I'll close this now.

Please take a look at the instructions on this page to install Docker using the official packages to get a supported stable setup: https://docs.docker.com/engine/installation/ubuntulinux/

The kernel you're running is also known to have data corruption bugs. You'll want to install all system updates.

Please feel free to comment. There's no need to open a new issue.
